.github/workflows/docker-build-test.yaml

```diff
@@ -307,17 +301,116 @@ jobs:
         github.event.pull_request.auto_merge != null ||
         contains(github.event.pull_request.body, '#e2e')
       )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all branches and tags
+
+      # To test the compat test against the last released docker image, we need to get the last released docker image tag.
+      # This step determines appropriate branch to fetch the latest docker image tag from based on the current branch.
+      # Determine the IMAGE_TAG based on the latest release R and check for images built on branch R - 1:
+      # - ex. For commits on the aptos-release-v1.19 branch, the IMAGE_TAG should be the latest commit built on aptos-release-v1.18
+      # - For commits on the main branch, the IMAGE_TAG should be the latest commit on the max release aptos-release-v<MAX> branch
+      - name: Determine branch to fetch latest docker image tag from
+        id: determine-the-target-branch
+        run: |
+          # Determine the base branch (the branch we're merging into)
+          if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "pull_request_target" ]]; then
+            base_branch="${{ github.base_ref }}"
+          else
+            base_branch="${{ github.ref_name }}"
+          fi
+
+          if [[ $base_branch == "main" ]]; then
+            target_branch=$(git ls-remote --heads origin 'aptos-release-v*' | 
```

Comment:
you probably wanna quote "$(git ...)" because any spaces there will break your output